src/modules/link-selector.js

```diff
@@ -14,7 +14,25 @@
 
 	var update = function () {
 		var currentPath = window.location.pathname;
-		scope.find('a[href]').each(function () {
+		var links = scope.find('a[href]');
+
+		links.each(function () {
+			var t = $(this);
+
+			App.modules.notify('changeState.update', {
+				item: t,
+				state: 'current-link-partial',
+				action: 'off'
+			});
+	
+			App.modules.notify('changeState.update', {
+				item: t,
+				state: 'current-link',
+				action: 'off'
+			});
+		});
+
+		links.each(function () {
```

Comment:
I am really not sure about this one, since it will create lots of layout trashing (possibility of `links.length * 4` to be exact)...

Why is it needed ?

Reply:
After discussion with @PascalPiche, we will be able to remove the layout trashing with the new fx.